application/console/migrations/m170203_210542_create_initial_tables.php

```diff
@@ -39,38 +42,44 @@ private function createUserTable()
             "ENGINE=InnoDB DEFAULT CHARSET=utf8"
         );
 
-        $this->createIndex('uq_user_employee_id','{{user}}','employee_id',true);
-        $this->createIndex('uq_user_username','{{user}}','username',true);
-        $this->createIndex('uq_user_email','{{user}}','email',true);
+        $this->createIndex('uq_user_employee_id', '{{user}}', 'employee_id', true);
+        $this->createIndex('uq_user_username', '{{user}}', 'username', true);
+        $this->createIndex('uq_user_email', '{{user}}', 'email', true);
     }
 
     private function dropUserTable()
     {
         $this->dropTable('{{user}}');
     }
 
-    private function createPasswordHistoryTable()
+    private function createPasswordTable()
     {
         $this->createTable(
-            '{{password_history}}',
+            '{{password}}',
             [
                 'id' => 'pk',
                 'user_id' => 'int(11) not null',
-                'password_hash' => 'varchar(255) not null',
+                'hash' => 'varchar(255) not null',
                 'created_utc' => 'datetime not null',
+                'expiration_utc' => 'datetime not null',
+                'grace_period_ends_utc' => 'datetime not null',
             ],
             "ENGINE=InnoDB DEFAULT CHARSET=utf8"
         );
+    }
 
-        $this->createIndex('idx_password_hash','{{password_history}}','password_hash',false);
-
-        $this->addForeignKey('fk_password_history_user_id','{{password_history}}','user_id','{{user}}','id','CASCADE','CASCADE');
+    private function dropPasswordTable()
+    {
+        $this->dropTable('{{password}}');
     }
 
-    private function dropPasswordHistoryTable()
+    private function createForeignKeys()
     {
-        $this->dropForeignKey('fk_password_history_user_id','{{password_history}}');
+        $this->addForeignKey('fk_user_to_current_password','{{user}}','password_id','{{password}}','id','CASCADE','CASCADE');
```

Comment:
Will we also want a foreign key from `password.user_id` to `user.id`? I don't actually know, since it's possible that having the foreign key going both directions will presumably affect the relationships Yii provides between those two models, and it might not properly infer them the way we have in mind.